### ISSUE TITLE
Cache and CORS middleware conflict

### ISSUE THREAD

**Issue by captainjapeng** [honojs/hono#963]
Hi!

I'm using both Cache middleware w/ the CORS middleware and there is a conflict. Cloudflare Worker throws this error `Can't modify immutable headers` when the cached response is returned directly.

As a workaround I'm using the ff code:
```ts
  const cacheKey = new Request(ctx.req.url.toString(), ctx.req.raw)
  const cache = await caches.open('test')
  const cacheResp = await cache.match(cacheKey)
  if (cacheResp) return new Response(cacheResp.body, cacheResp)
  ```

**Comment by yusukebe** [comment]
Hi @captainjapeng !

It may be fixed in the next version (v3.1). I'll let you know if it works or not. Please wait!

**Comment by yusukebe** [comment]
This has been fixed. If you have any problems related to this issue, feel free to reopen it.
